common/src/worldgen.rs

```diff
@@ -130,19 +130,75 @@ impl NodeState {
     }
 }
 
+type VoxelCoords = (u8, u8, u8);
+
+struct VoxelIterable {
+    coords: VoxelCoords,
```

Comment:
This would be much simpler if it was just a `counter: u32`, with `Iterator::next` being something like
```rust
if self.counter == u32::from(self.dimension).pow(3) { return None; }
let result = (self.counter % self.dimension, self.counter / self.dimension, self.counter / self.dimension.pow(2));
self.counter += 1;
Some(result)
```

Reply:
Made this change. 

Is there a way I could have avoided the conversion from u8 to u32 and back to u8? I always feel like I've painted myself into a corner whenever I end up having to do type conversions just for a calculation in one place, but I don't want to do things that may have far-reaching consequences such as changing the type of ChunkParams.dimension.